optimum/gptq/quantizer.py

```diff
@@ -103,6 +122,14 @@ def __init__(
                 Whether to perform sequential quantization even within a single Transformer block.
                 Instead of quantizing the entire block at once, we perform layer-wise quantization.
                 As a result, each layer undergoes quantization using inputs that have passed through the previously quantized layers.
+            checkpoint_format (`str`, *optional*, defaults to `gptq`):
+                GPTQ weight format. `gptq`(v1) is supported by both gptqmodel and auto-gptq. `gptq_v2` is gptqmodel only.
+            meta (`Dict[str, any]`, *optional*):
+                Properties, such as tooling:version, that do not directly contributes to quantization or quant inference are stored in meta.
+                i.e. `meta.quantizer`: ["optimum:_version_", "gptqmodel:_version_"]
+            backend (`str`, *optional*):
+                Controls which gptq kernel to be used. Valid values for gptqmodel are `auto`, `auto_trainable` and more. For auto-gptq, only 
+                valid value is None and `auto_trainable`. Ref gptqmodel backends: https://github.com/ModelCloud/GPTQModel/blob/main/gptqmodel/utils/backend.py
```

Comment:
@SunMarc This is the biggest change and one I need to go into explicit details since they are non-obvious. 

`checkpoint_format` is added to both auto-gptq(main) but never released and carried-over to gptqmodel since I see as a good addition since the method of gptq produces many kernels, each may use a specific weight/disk format. 

`meta` is added by gptqmodel to store `info` only properties that is not related to the loading and execution of the quantized model. Most importantly, it stores a `meta.quantizer`  property [list of: quantizer:version]  tooling that produced the quant. This is extremely valuable for two reasons:

1. [Good to have but not essential] Debugging and tracing back bad quants generated by bad code/tools. Who made the (bad/good) quants? This is an tooling fingerprint since there are multiple tools that can produce gptq format. They are not equal and this will allow everyone to trace to origins. In this PR, meta.quantizer is size 2 array holding both optimum version + (gptqmodel version or autogptq version)

2. [Requirement for GPTQMode + future bug proofing] Backward compat and future bug-proofing. GPTQModel uses this to test of a zeropoint fix made by @qwopqwop200 that affects all `gptq` (v1) disk format created before this fix. Models made before this fix has broken `sym=false` zeropoint. Models quantized after this fix can load `sym=False`. This is a safety check since there are two version of gptq v1 that is compatible with eather when sym=true, and only after fix for sym=false. 

3. [Essential for GPTQModel and also good for auto-gptq] `backend`. The old auto-gptq method of selecting which kernel/quant_linear for which task/model is extremely cryptic and controlled by 3 params `disable_exllama`, `exllama_config`, `disable_exllama` plus 1 code state called `use_exllama`. Frankly, this control scheme no longer makes logical sense and is borderline crazy. GPTQmodel uses a single `backend` to signal kernel selection. 

The kernel selection comes down to this logic split into two core paths: Does the model require training? aka will it enter peft path. True, than select the best kernel that can be trained on. False, select a best kernel for quant/inference.  Existing gptqmodel `checkpoint_formats` are `gptq`, `gptq_v2`, `marlin`, `ipex`, `bitblas` with more coming this year. 

You can ignore all the switches as the basic need for the 3+1 variables in auto-gptq boils down to this. While trying to give users the ability to choose a specific kernel but I can safely say, there are maybe 3 people in the world that can select the correct kernel using auto-gptq kernel using the control maze. There are more toggle beyond 3+1 within autogptq. 

Due to the above, GPTQModel will not accept or adapt to to auto-gptq kernel selection crazyiness. 1 clean variable is all you need with two primar/auto states: `auto`, `auto_trainable`, plus individual kernels you can explicitly call. This is the best and only way out the mess of kernel selection. There are currently [8 kernels](https://github.com/ModelCloud/GPTQModel/tree/main/gptqmodel/nn_modules/qlinear) in gptqmodel with more coming this year plus even more checkpoint_formats beyond gptq, gptq_v2, marlin, ipex, bitblas.  We are not going down the auto-gptq path of adding a 1970 telco phoneline switch style variable for each kernel. 

But, for the sake of compatibility, this PR contains code that will allow users to pass only auto-gptq control vars and convert that to gptqmodel auto loading state of `auto` and `auto_trainable` and reversely, allow passing `backend=auto_trainable` and to auto-gptq kernel selection control. 

I apologize for the overly verbose message here but this is the meat of the pr as far as potential friction for review since I can totally see why anyone seeing these changes will throw blank stares without me explaining every single detail about each new param/state var.